### PR TITLE
[PlatformIndicators] Add own Status

### PR DESCRIPTION
**Pull request by HypedDomi**
4XnVG1

### CODE REVIEW

**Comment by Vendicated** [comment]
lol whats the point of this, u should know which platform youre on

**Comment by JoshHyde9** [comment]
lol what is the point of you

**Comment by HypedDomi** [comment]
> lol whats the point of this, u should know which platform youre on

Yeah idk. Just felt it would fit to the plugin

**Comment by voidfill** [comment]
major L

**Comment by Vendicated** [comment]
as far as I can tell, wouldn't this be unable to differentiate between multiple online sessions? I'd rather not have a feature than have it but broken

**Comment by HypedDomi** [comment]
> as far as I can tell, wouldn't this be unable to differentiate between multiple online sessions (if you're online on say mobile and desktop at the same time)? I'd rather not have a feature than have it but broken

It can. getSessions returns the client and the status for the session
image
image

**Comment by Vendicated** [comment]
lol
image

**Comment by HypedDomi** [comment]
> lol image

That combines every session. still no clue why but it filters this out

File: src/plugins/platformIndicators.tsx [hunk 1]
```diff
@@ -62,6 +64,30 @@ const PlatformIcon = ({ platform, status }: { platform: Platform, status: string
...
+        if (!sessions) return null;
+
+        if (typeof sessions !== "object") return null;
+        const sortedSessions = Object.values(sessions).sort((a: any, b: any) => {
+            if (a.status === "online" && b.status !== "online") return 1;
+            if (a.status !== "online" && b.status === "online") return -1;
+            if (a.status === "idle" && b.status !== "idle") return 1;
+            if (a.status !== "idle" && b.status === "idle") return -1;
+            return 0;
+        });
```
Comment (Vendicated) [inline src/plugins/platformIndicators.tsx:79]: so what's the point of this sort?

Comment (HypedDomi) [reply src/plugins/platformIndicators.tsx:79]: This sorts the status. So it doesnt show for example idle for your desktop when youre online and on another pc in idle

File: src/plugins/platformIndicators.tsx [hunk 2]
```diff
@@ -20,10 +20,12 @@ import { Settings } from "@api/settings";
 import ErrorBoundary from "@components/ErrorBoundary";
 import { Devs } from "@utils/constants";
 import definePlugin, { OptionType } from "@utils/types";
-import { findByCodeLazy } from "@webpack";
-import { PresenceStore, Tooltip } from "@webpack/common";
+import { findByCodeLazy, findByProps } from "@webpack";
+import { PresenceStore, Tooltip, UserStore } from "@webpack/common";
 import { User } from "discord-types/general";
 
+let SessionStore = findByProps("getActiveSession");
```
[the remark this answers is not in the document]
Comment (Vendicated) [reply src/plugins/platformIndicators.tsx:27]: image
oh my god i hate github i forgot to submit this review

**Review by Vendicated — APPROVED** [review]

**Comment by Vendicated** [comment]
these alignment fixes aren't related to ur pr but i just pushed them here anyway

**Pull request merged**